contracts/eos/BancorNetwork/BancorNetwork.hpp

```diff
@@ -25,35 +25,85 @@ using namespace std;
  * - For example, in order to convert 10 EOS into BNT, the caller needs to transfer 10 EOS to the contract
  * and provide the following memo:
  * > `1,bnt2eoscnvrt BNT,1.0000000000,receiver_account_name`
+ * - Optionally, an affiliate fee and affiliate account may be included (either both or neither) as such:
+ * > `1,PATH,1.0000000000,receiver_account_name,affiliate_account_name,affiliate_fee`
  * @{
 */
 
+/**
+ * @details 
+ * - the account which originated the entire conversion path
+ * - the final destination of the entire conversion path
+ * - the contract where this conversion was processed
+ * - the return from the conversion in BNT before fee deduction
+ * - the account that was paid the affiliate fee
+ * - the amount of that was paid as affiliate fee 
+ */
+#define EMIT_AFFILIATE_EVENT(sender, destination, from_contract, fee_account, to_amount, fee_amount){ \
+    START_EVENT("affiliate", "1.0") \
+    EVENTKV("sender", sender) \
+    EVENTKV("destination", destination) \
+    EVENTKV("from_contract", from_contract) \
+    EVENTKV("affiliate_account", fee_account) \
+    EVENTKV("return", to_amount) \
+    EVENTKVL("affiliate_fee", fee_amount) \
+    END_EVENT() \
+}
+
 /*! \cond DOCS_EXCLUDE */
 CONTRACT BancorNetwork : public eosio::contract { /*! \endcond */
     public:
         using contract::contract;
 
-        ACTION init();
+        /**
+         * @defgroup Network_Settings_Table Settings Table
+         * @brief This table stores the settings for the entire Bancor network
+         * @{
+         *//*! \cond DOCS_EXCLUDE */
+            TABLE settings_t { /*! \endcond */
+                /**
+                 * @brief conversions through all the converters in the Bancor network may be disabled with this toggle
+                 */
+                bool enabled;
```

Comment:
I can remove the setenabled action which will ignore this, but we need to keep the variable in the struct, because we also reference the struct in a different context (when trying to read converter settings) in `isConverter`

Reply:
how is that related? can't we just remove the entire `enabled` logic, remove it from the struct and keep everything else in the struct?